UPGRADE-3.3.md

```diff
@@ -82,6 +82,8 @@ DependencyInjection
 
  * Autowiring services based on the types they implement is deprecated and won't be supported in version 4.0. Rename (or alias) your services to their FQCN id to make them autowirable.
 
+ * [BC BREAK] The `NullDumper` class has been made final
```

Comment:
`@final` + the BC break means to me that we've considered the class final for real, taking advantage of it. To me, it sends the good info to the reader.